src/onemod/fsutils/data_loader.py

```diff
@@ -30,24 +30,17 @@ def load(
             raise ValueError(f"Unsupported data format for '{path.suffix}'")
 
         if return_type == "pandas_dataframe":
-            pandas_df = self.io_dict[path.suffix].load_eager(path, **options)
-            assert isinstance(
-                pandas_df, pd.DataFrame
-            ), "Expected a pandas DataFrame"
+            pandas_df = self.io_dict[path.suffix].load_eager(
+                path, columns=columns, subset=subset, **options
+            )
+            assert isinstance(pandas_df, pd.DataFrame), (
+                "Expected a pandas DataFrame"
+            )
 
+            # Enforce `column` order if requested and reset index in case of `subset`.
             if columns:
                 pandas_df = pandas_df[columns]
```

Comment:
If we deal with the columns parameters in the detailed `_impl` function, I think we can skip the `columns` logic here.